src/ParticleDA.jl

```diff
@@ -384,35 +399,9 @@ function update_particle_proposal!(model_data, filter_data, truth_observations,
 end
 
 function update_particle_proposal!(model_data, filter_data, truth_observations, nprt_per_rank, filter_type::OptimalFilter)
-
-        # Optimal Filter: After updating the particle dynamics, we apply the "optimal proposal" in
-        #                 sample_height_proposal!() to the first state variable (height). We apply
-        #                 a sample from the gaussian random field in update_particle_noise!() to the other
-        #                 state variables (velocity).
-
-        particles = get_particles(model_data)
-
-        # Apply optimal proposal, the result will be in offline_matrices.samples
-        sample_height_proposal!(@view(particles[:,:,1,:]),
-                                filter_data.offline_matrices,
-                                filter_data.online_matrices,
-                                truth_observations,
-                                filter_data.stations,
-                                filter_data.grid,
-                                filter_data.grid_ext,
-                                filter_data.fft_plan,
-                                filter_data.fft_plan!,
-                                nprt_per_rank,
-                                filter_data.rng[threadid()],
-                                filter_data.obs_noise_std)
-
-        # Add noise from the standard gaussian random field to all state variables in model_data
-        update_particle_noise!(model_data, nprt_per_rank)
-
-        # Overwrite the height state variable with the samples of the optimal proposal
-        particles[:,:,1,:] .= filter_data.online_matrices.samples
-        set_particles!(model_data, particles)
-
+    update_particle_noise!(model_data, nprt_per_rank)
+    indices = get_indices(model_data)
```

Comment:
`indices` is unused?

Reply:
Yes, this has been removed. At the time I was thinking of just passing the observed state variables to update_particles_given_observations!